### PR TITLE
[release-4.3] BUG 1798344: Optimize VF configure logic

### PR DESCRIPTION
**Pull request by pliurh**
Check whether desired state is the same as the current one before
executing.

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -131,10 +131,12 @@ func SyncNodeState(newState *sriovnetworkv1.SriovNetworkNodeState) error {
 }
 
 func needUpdate(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetworkv1.InterfaceExt) bool {
-	if iface.Mtu != ifaceStatus.Mtu {
+	if iface.Mtu > 0 && iface.Mtu != ifaceStatus.Mtu {
```
Comment (zshi-redhat) [inline pkg/utils/utils.go]: The range of `iface.Mtu` is defined in crd policy definition, do we still need to check '>0' ? Similar to `iface.NumVfs`, we didn't check if it's larger than 0.

Comment (pliurh) [reply pkg/utils/utils.go]: The MTU can be omitted in policy, in such case, golang will assign a default value '0' to the int variable.  So if user doesn't specify the MTU in policy explicitly, the configure daemon shall not set the MTU. 

For NumVfs, we shall reset the PF to 0 VF, when user set the NumVfs to 0 in policy either explicitly or implicitly.

**Pull request merged**